parsers/ocaml.c

```diff
@@ -1864,25 +2067,55 @@ static void findOcamlTags (void)
 	lexingState st;
 	ocaToken tok;
 
+	/* One-token lookahead gives us the ability to
+	 * do much more accurate analysis */
+	lexingState nextSt;
+	ocaToken nextTok;
+
 	initStack ();
+
 	computeModuleName ();
 	tempIdent = vStringNew ();
 	lastModule = vStringNew ();
 	lastClass = vStringNew ();
 	voidName = vStringNew ();
 	vStringCopyS (voidName, "_");
+	vString *temp_cp = vStringNew ();
 
-	st.name = vStringNew ();
-	st.cp = fileReadLine ();
+	nextSt.name = vStringNew ();
+	nextSt.cp = fileReadLine ();
 	toDoNext = &globalScope;
-	tok = lex (&st);
+	nextTok = lex (&nextSt);
+
+	/* prime the lookahead token */
+	st = nextSt;	// preserve the old state for our first token
+	st.name = vStringNewCopy (st.name);
+	st.cp = (const unsigned char *) temp_cp->buffer;
```

Comment:
shouldn't the buffer be copied from `nextSt.cp` first like [below](#diff-9ee068438f29821d9f8a0cea175b87acR2110)?